### PR TITLE
For #14609, Add `SHOW UNUSED SHARDING ALGORITHMS` syntax to DistSQL.

### PR DESCRIPTION
**Pull request by yx9o**
For #14609, Add `SHOW UNUSED SHARDING ALGORITHMS` syntax to DistSQL.

Changes proposed in this pull request:

Add `SHOW UNUSED SHARDING ALGORITHMS` syntax.
Add unit test.

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/UnusedShardingAlgorithmQueryResultSet.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Result set for show unused sharding algorithm.
```
Comment (RaigorJiang) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/UnusedShardingAlgorithmQueryResultSet.java:43]: algorithm

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/UnusedShardingAlgorithmQueryResultSet.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ */
+public final class UnusedShardingAlgorithmQueryResultSet implements DistSQLResultSet {
```
Comment (RaigorJiang) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/UnusedShardingAlgorithmQueryResultSet.java:45]: Algorithms, not Algorithm

Comment (yx9o) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/UnusedShardingAlgorithmQueryResultSet.java:45]: > Algorithms, not Algorithm

Ok.

Comment (yx9o) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/UnusedShardingAlgorithmQueryResultSet.java]: Done, please review, thank you.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**